File: 0008/P0008.cpp
#include <algorithm>
#include <iostream>
#include <functional>
#include <vector>

int count(const std::vector<int>& numbers, int k) {
  int maxProduct = 0;
  for (int i = 0; i < numbers.size() - k; ++i) {
    int product = 1;
    for (int j = i; j < i + k; ++j) {
      product *= numbers[j];
    }
    maxProduct = std::max(maxProduct, product);
  }

  return maxProduct;
}

int main() {
  int t;
  std::cin >> t;
  for (int i = 0; i < t; ++i) {
    int n, k;
    std::cin >> n >> k;
    std::string s;
    std::cin >> s;
    std::vector<int> numbers;
    std::transform(s.begin(), s.end(), std::back_inserter(numbers), [&] (const char& c) {
      return int(c - '0');
    });

    int result = count(numbers, k);

    std::cout << result << std::endl;
  }
}

File: 0008/P0008.py
from functools import reduce
from operator import mul

def count(numbers, k):
  max_product = 0
  for i in range(len(numbers) - k):
    product = reduce(mul, [ numbers[i] for i in range(i, i + k)])
    max_product = max(max_product, product)
  return max_product

if __name__ == "__main__":
  t = int(input())
  for i in range(t):
    n, k = map(int, input().split())
    numbers = list(map(int, list(input())))
    result = count(numbers, k)
    print(int(result))